python/py_helpers.test.py

```diff
@@ -1018,7 +1159,146 @@ def test_has_import(self):
+            node.find_function("foo")
+            .find_return()
+            .find_comp_iters()[0]
+            .is_equivalent("spam")
+        )
+        self.assertTrue(
+            node.find_function("foo")
+            .find_return()
+            .find_comp_iters()[1]
+            .is_equivalent("lst")
+        )
+
+    def test_find_comp_targets(self):
+        code_str = """
+x = [i**2 for i in lst]
+
+def foo(spam):
+  return [i * j for i in spam for j in lst]
+"""
+        node = Node(code_str)
+
+        self.assertEqual(len(node.find_variable("x").find_comp_targets()), 1)
+        self.assertTrue(
+            node.find_variable("x").find_comp_targets()[0].is_equivalent("i")
+        )
+        self.assertEqual(
+            len(node.find_function("foo").find_return().find_comp_targets()), 2
+        )
+        self.assertTrue(
+            node.find_function("foo")
+            .find_return()
+            .find_comp_targets()[0]
+            .is_equivalent("i")
+        )
+        self.assertTrue(
+            node.find_function("foo")
+            .find_return()
+            .find_comp_targets()[1]
+            .is_equivalent("j")
+        )
+
+    def test_find_comp_key(self):
+        code_str = """
+x = {k: v for k,v in dict}
+
+def foo(spam):
+  return {k: v for k in spam for v in lst}
+"""
+        node = Node(code_str)
+
+        self.assertTrue(node.find_variable("x").find_comp_key().is_equivalent("k"))
+        self.assertTrue(
+            node.find_function("foo").find_return().find_comp_key().is_equivalent("k")
+        )
+
+    def test_find_comp_expr(self):
+        code_str = """
+x = [i**2 if i else -1 for i in lst]
+
+def foo(spam):
+  return [i * j for i in spam for j in lst]
+"""
+        node = Node(code_str)
+
+        self.assertTrue(
+            node.find_variable("x").find_comp_expr().is_equivalent("i**2 if i else -1")
+        )
+        self.assertTrue(
+            node.find_function("foo")
+            .find_return()
+            .find_comp_expr()
+            .is_equivalent("i*j")
+        )
+
+    def test_find_comp_ifs(self):
+        code_str = """
+x = [i**2 if i else -1 for i in lst]
+
+def foo(spam):
+  return [i * j for i in spam if i for j in lst if j]
+"""
+        node = Node(code_str)
+
+        self.assertEqual(len(node.find_variable("x").find_comp_ifs()), 0)
+        self.assertEqual(
+            len(node.find_function("foo").find_return().find_comp_ifs()), 2
+        )
+        self.assertTrue(
+            node.find_function("foo")
+            .find_return()
+            .find_comp_ifs()[0]
+            .is_equivalent("i")
+        )
+        self.assertTrue(
+            node.find_function("foo")
+            .find_return()
+            .find_comp_ifs()[1]
+            .is_equivalent("j")
+        )
+
+
 class TestGenericHelpers(unittest.TestCase):
+    def test_has_node(self):
+        self.assertTrue(Node("self.matrix[1][5] = 3").has_node("self.matrix[1][5] = 3"))
```

Comment:
```suggestion
        self.assertTrue(Node("name = input('hi')\nself.matrix[1][5] = 3").has_stmt("self.matrix[1][5] = 3"))
```
At first glance I wasn't sure why `has_node` exists, when `is_equivalent` would have worked here. Adding an extra statement makes it obvious that's not the case.

Also shouldn't it be `has_statement` or `has_stmt`?